src/classes/groundplaces.ts

```diff
@@ -201,31 +194,119 @@ export class GroundPlacesController {
     intoStopGroupGpuid: StopGroupGpuid,
   ): void {}
 
+  /**
+   * @description Add a stopGroup to a stopCluster.
+   * @param {StopGroupGpuid} stopGroupGpuidToAdd - Ground place unique identifier of the StopGroup to add.
+   * @param {StopClusterGpuid} intoStopClusterGpuid - Ground Place unique identifier of the new StopCluster parent.
+   * @returns {void}
+   */
+  public addStopGroupToStopCluster(stopGroupGpuidToAdd: StopGroupGpuid, intoStopClusterGpuid: StopClusterGpuid): void {
+    // Check if the StopGroup specified exist in the Ground places
+    this.storageService.getStopGroupByGpuid(stopGroupGpuidToAdd);
+
+    const stopClusterParent: StopCluster = this.storageService.getStopClusterByGpuid(intoStopClusterGpuid);
+
+    // Check if the StopGroup does not already exist in the new StopCluster parent
+    const isStopGroupAlreadyExist = Boolean(
+      stopClusterParent.childs.find((stopGroupGpuid: StopGroupGpuid) => stopGroupGpuid === stopGroupGpuidToAdd),
+    );
+
+    if (isStopGroupAlreadyExist) {
+      throw new Error(
+        `The StopGroup with the Gpuid "${stopGroupGpuidToAdd}" cannot be added to the StopCluster with the Gpuid "${intoStopClusterGpuid}" because it already exist in it.`,
+      );
+    }
+
+    stopClusterParent.childs.push(stopGroupGpuidToAdd);
+
+    // Update the place with the new StopCluster parent
+    this.storageService.replacePlace(stopClusterParent);
+  }
+
+  /**
+   * @description Remove a stopGroup from a stopCluster.
+   *
+   * Warning: The StopGroup cannot be without StopCluster parent after this operation.
+   * @param {StopGroupGpuid} stopGroupGpuidToRemove - Ground place unique identifier of the SopGroup to remove.
+   * @param {StopClusterGpuid} stopClusterGpuidParent - Ground place unique identifier of the StopCluster parent.
+   * @returns {void}
+   */
+  public removeStopGroupFromStopCluster(
+    stopGroupGpuidToRemove: StopGroupGpuid,
+    stopClusterGpuidParent: StopClusterGpuid,
+  ): void {
+    const stopClusterParent: StopCluster = this.storageService.getStopClusterByGpuid(stopClusterGpuidParent);
+    const groundPlaces: GroundPlace[] = this.storageService.getGroundPlaces();
+    const copyGroundPlaces: GroundPlace[] = this.storageService.cloneGroundPlaces();
```

Comment:
have you tried using lodash deepClone/deepCopy here instead of a storage method that `parse(stringify())` ?

Reply:
Yes, use with lodash.clonedeep module